src/index.js

```diff
@@ -129,7 +130,19 @@ var JobQueueService = function (storage, type) {
           logService.error(err);
         }
       }
-      processJob(job, complete);
+      processJob(job, function(){
+        var jobCompleteTime = new Date().getTime();
+        var jobCompletionTime = jobCompleteTime - jobStartTime;
+        let trackingEvent = {
+          _type: 'TrackingEvent',
+          eventName: job.type + '-job-complete',
+          happenedAt: new Date(),
+          eventData: { jobCompletionTime: jobCompletionTime },
+        }
+        dataService.create(trackingEvent).then(function(data){
```

Comment:
@kimobrian the complete callback shouldn't wait until the event is created (that adds unnecessary delays to our job completion times & the UI waits for these jobs in many cases)

Instead, we want:
dataService.create(trackingEvent);
complete();

Make sense?

Reply:
Yap